src/components/RoundForm.vue

```diff
@@ -0,0 +1,264 @@
+      courses: [],
+      newCourse: {},
+      newPlayer: '',
+      clubOptions: [],
+      courseOptions: [],
+      playerOptions: [],
+    }
+  },
+  computed: {
+    newCourseOptions () {
+      let existingCourses = this.courses.map(c => c.name)
+      return this.courseOptions.filter(c => !existingCourses.includes(c.name))
+    },
+    canAddCourses () {
+      return this.courses.length < 2
+    },
+    newPlayerOptions () {
+      if (this.courses.length == 0) {
+        return []
+      }
+      let players = this.courses[0].scores.map(s => s.player)
+      return this.playerOptions.filter(p => !players.includes(p))
+    },
+    canAddScores () {
+      return this.courses[0].scores.length < 4
+    },
+    submitWarningMsg () {
+      if (!this.date) return 'Missing date'
+      if (Object.keys(this.club).length == 0) return 'Missing club'
+      if (this.courses.length == 0)  return 'Missing course'
+      if (!this.courses.every(c => c.tees)) return 'Missing tees'
+      if (!this.courses.every(c => c.numHoles > 0)) return 'Missing holes'
+      if (!this.courses.every(c => c.scores.length > 0)) {
+        return 'Missing players'
+      }
+      let strokes = this.courses.flatMap(c => c.scores.flatMap(s => s.strokes))
+      if (strokes.includes(0)) {
+       return 'Missing scores. All scores must be greater than 0'
+      }
+      return ''
+    },
+  },
+  created () {
+    this.$http.get(
+      '/api/input-rounds'
+    ).then(function (response) {
+      this.clubOptions = response.data.clubs;
+      this.playerOptions = response.data.players;
+      this.msg = 'Input new round below.'
+    }, function (error) {
+      console.log(error.data);
+      alert(error.data);
+    })
+  },
+  methods: {
+    changeClub: function() {
+      this.courses = []
+      this.courseOptions = this.club.courses
+    },
+    addCourse: function() {
+      this.courses.push({
+        name: this.newCourse.name,
+        teesOptions: this.newCourse.tees,
+        tees: '',
+        numHoles: 0,
+        scores: [],
+      })
+      this.newCourse = {}
+    },
+    removeCourse: function(index) {
+      this.courses.splice(index, 1)
+    },
+    changeNumHoles(courseIndex) {
+      let course = this.courses[courseIndex]
+      if (course.scores.length == 0) {
+        return
+      }
+      let oldStrokesLength = course.scores[0].strokes.length
+      let diff = course.numHoles - oldStrokesLength
+      if (diff > 0) {
+        let strokes = Array(diff).fill(0)
+        course.scores.forEach(s => s.strokes.push(...strokes))
+      } else {
+        // start splicing after the "numHoles" index, so numHoles + 1,
+        // but numHoles is 1-indexed, so start at numHoles
+        // Ex. numHoles = 3, start splicing at index 3
+        let startIndex = course.numHoles
+        course.scores.forEach(s => s.strokes.splice(startIndex))
+      }
+    },
+    addScore: function() {
+      for (let course of this.courses) {
+        course.scores.push({
```

Comment:
`course.scores` is new, the old code was just storing `course.strokes` and it was storing `players` outside of the `course`.  The old code had a separate select for players outside the strokes table. The new code simplifies the UI by letting the user manage players in the strokes table.